lib/plan/setting_panel/setting_panel.dart

```diff
@@ -0,0 +1,284 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:trufi_core/widgets/custom_expanded_tile.dart';
+import 'package:trufi_core/widgets/custom_switch_tile.dart';
+
+import '../../entities/plan_entity/plan_entity.dart';
+
+import 'setting_panel_cubit.dart';
+
+class SettingPanel extends StatelessWidget {
+  static const String route = "/setting-panel";
+  const SettingPanel({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    // TODO to add all translates this widget
+    // final localization = TrufiLocalization.of(context);
+
+    final settingPanelCubit = context.read<SettingPanelCubit>();
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Settings"),
+        leading: BackButton(
+          onPressed: () {
+            Navigator.of(context).pop(settingPanelCubit.state);
+          },
+        ),
+      ),
+      body: SafeArea(
+        child: BlocBuilder<SettingPanelCubit, SettingPanelState>(
+          builder: (blocContext, state) {
+            return ListView(
+              children: <Widget>[
+                const SizedBox(
+                  height: 10,
+                ),
+                CustomExpansionTile(
+                  title: "Walking speed",
+                  options: const [
+                    'Slow',
+                    'Calm',
+                    'Average',
+                    'Prompt',
+                    'Fast',
+                  ],
+                  textSelected: _getTextWalkingSelect(context,state.typeWalkingSpeed),
+                  onChanged: (value) {
```

Comment:
If the option is the enum you can just call the settingPanelCubit.setWalkingSpeed(value)

Reply:
Thanks :) , it is changed